From 92827c69197d5700a96438365999fc7feafc060f Mon Sep 17 00:00:00 2001
From: australiensun <australien@sun>
Date: Sat, 16 Nov 2013 11:16:58 +0100
Subject: [PATCH] progress_indicator.cpp: fix by using wxString() ctor

(cherry picked from commit e79bdcae9811ea7aa9bc7961a81fdc33f0c071fe)
---
 FreeFileSync/Source/ui/progress_indicator.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/FreeFileSync/Source/ui/progress_indicator.cpp b/FreeFileSync/Source/ui/progress_indicator.cpp
index 0bfd238..dd24637 100644
--- a/FreeFileSync/Source/ui/progress_indicator.cpp
+++ b/FreeFileSync/Source/ui/progress_indicator.cpp
@@ -1084,7 +1084,7 @@ struct LabelFormatterTimeElapsed : public LabelFormatter
     {
         if (!drawLabel_) return wxString();
         return timeElapsed < 60 ?
-               _P("1 sec", "%x sec", numeric::round(timeElapsed)) :
+               wxString(_P("1 sec", "%x sec", numeric::round(timeElapsed))) :
                timeElapsed < 3600 ?
                wxTimeSpan::Seconds(timeElapsed).Format(   L"%M:%S") :
                wxTimeSpan::Seconds(timeElapsed).Format(L"%H:%M:%S");
@@ -1869,8 +1869,8 @@ void SyncProgressDialogImpl<TopLevelDialog>::processHasFinished(SyncResult resul
             //set overall speed (instead of current speed)
             const int64_t timeDelta = timeElapsed.timeMs() - phaseStartMs; //we need to consider "time within current phase" not total "timeElapsed"!
 
-            const wxString overallBytesPerSecond = timeDelta == 0 ? wxString() : filesizeToShortString(dataCurrent * 1000 / timeDelta) + _("/sec");
-            const wxString overallItemsPerSecond = timeDelta == 0 ? wxString() : replaceCpy(_("%x items/sec"), L"%x", formatThreeDigitPrecision(itemsCurrent * 1000.0 / timeDelta));
+            const wxString overallBytesPerSecond = timeDelta == 0 ? wxString() : wxString(filesizeToShortString(dataCurrent * 1000 / timeDelta) + _("/sec"));
+            const wxString overallItemsPerSecond = timeDelta == 0 ? wxString() : wxString(replaceCpy(_("%x items/sec"), L"%x", formatThreeDigitPrecision(itemsCurrent * 1000.0 / timeDelta)));
 
             pnl.m_panelGraphBytes->setAttributes(pnl.m_panelGraphBytes->getAttributes().setCornerText(overallBytesPerSecond, Graph2D::CORNER_TOP_LEFT));
             pnl.m_panelGraphItems->setAttributes(pnl.m_panelGraphItems->getAttributes().setCornerText(overallItemsPerSecond, Graph2D::CORNER_TOP_LEFT));
-- 
1.8.4

